index.js

```diff
@@ -100,62 +110,73 @@ function getStylesFromIframe(iframe) {
 }
 
 module.exports = function() {
+    var codepens = document.querySelectorAll('div.codepen');
+    //remove the old codepen links
+    codepens.forEach(function(codepen, i){
+        var wrapper = findSelector(codepen, "pre, .demo_wrapper");
+        //the CodePen iframe wrapper has ".codepen" class too
+        if (wrapper) {
+            wrapper.setAttribute('data-has-run', true);
+            codepen.parentNode.removeChild(codepen);
+        }
+    });
 
-    document.body.addEventListener("click", function(ev){
-        if(matches.call(ev.target, ".codepen")){
-
-            var el = findSelector(ev.target, "pre, .demo_wrapper");
-            if(el && matches.call(el, "pre")) {
-                var preElement = el;
-                var codeElement = preElement.querySelector("code");
-                var language = codeElement.className.match(languageHTML)[1];
-                var text = codeElement.textContent;
-
-                var data = types[language](text);
-
-                if(data.js) {
-                    data.js = data.js.trim();
-                }
-                if(data.html) {
-                    data.html = data.html.trim();
-                }
-                if(data) {
-                    cleanCodePenData(data);
-                    if(window.CREATE_CODE_PEN) {
-                        CREATE_CODE_PEN(data);
-                    } else {
-                        createCodePen(data);
+    //Register PrismJS "Run" custom button
+    Prism.plugins.toolbar.registerButton("run-code", function(env) {
+        var demoWrapper = findDemoWrapper(env.element);
+        var pre = env.element.parentElement;
+        var hasRunBtn = demoWrapper ? demoWrapper.getAttribute("data-has-run") : pre.getAttribute("data-has-run");
+        //prevent other demos without codepen link to register Run button
+        if (hasRunBtn) {
+            var btn = document.createElement("button");
+            btn.innerHTML = "Run";
+            document.body.addEventListener('click', function (ev) {
```

Comment:
This event listener only gets added once, right? It doesn’t get added for each button?

[I see the event delegation, which is nice, but I want to double check that the listener isn’t added multiple times. If it is, this logic should be moved outside the `registerButton` call.]

Reply:
@chasenlehara Fixed, thank you!